drivers/UARTSerial.h

```diff
@@ -157,6 +157,12 @@ class UARTSerial : private SerialBase, public FileHandle {
      */
     void set_data_carrier_detect(PinName dcd_pin, bool active_high = false);
 
+    /** Set the baud rate
+     *
+     *  @param baud   The baud rate
+     */
+    void set_baud(int baud);
```

Comment:
It's private because if any user messes with the underlying port while we're using it, it could break us - at least if they go anywhere near the core data functions.

This is a driver that uses the underlying serial port itself - I don't want to expose it to the users of this class so they can freely touch it. Making it a public base class would expose more than I want.

In this particular case "baud" could directly passes through, as we don't care (beyond maybe it being a good idea to drain the output buffer?).

But "set_flow_control" we would potentially want to implement ourself rather than pass through to provide XON/XOFF or manual HW control.

There's existing inconsistency on the "set_" prefix ("baud" vs "set_flow_control"), so it seems reasonable to unify on the "set_" to me.

General note - we are currently lacking is a "serial control" abstract interface for these things (baud, format, flow control, hup) - each of the buffered serial, the raw serial, and the serial mux should all be providing that as a common interface.